modules/caddyhttp/headers/headers.go

```diff
@@ -119,7 +119,7 @@ type HeaderOps struct {
 	Set http.Header `json:"set,omitempty"`
 
 	// Names of HTTP header fields to delete.
-	Delete []string `json:"delete,omitempty"`
+	Delete []DeleteHeader `json:"delete,omitempty"`
```

Comment:
I'm worried about this change though, because this would be a breaking change for any JSON users. Whatever the solution is, it should retain the fact that a string can be used to configure deleted headers, for backwards compatibility.

My suggestion might be to instead of `DeleteHeader`, make a `deleteRegexps map[string]*regexp.Regexp` field which retains the compiled regexps, by the header config string. On provision, if it's a regexp, then add the compiled regexp to the map. Then you would do a lookup in the map by the `.Delete` string at runtime to fetch the regexp if any, and if not then just delete as normal. This should keep backwards-compat, I think.

@mholt how does this sound to you?

Reply:
Makes sense. I agree with you.